packages/workflow/src/utils.ts

```diff
@@ -30,22 +30,25 @@ export const deepCopy = <T>(source: T): T => {
 	return clone;
 };
 // eslint-enable
-type ErrorMessage = { errorMessage: string };
-type FallbackValue<T> = { fallbackValue: T };
 
-export const jsonParse = <T>(
-	jsonString: string,
-	options: ErrorMessage | FallbackValue<T> | {} = {},
-): T => {
+type MutuallyExclusiveTypeError = 'TypeError: Only one of the two options is allowed';
+
+type MutuallyExclusive<T, U> =
+	| (T & { [k in Exclude<keyof U, keyof T>]?: MutuallyExclusiveTypeError })
```

Comment:
I prefer `never` over `MutuallyExclusiveTypeError`, as the error thrown by `tsc` is equally confusing with either option, but with `never` you don't need to force case `fallbackValue`.

Reply:
I decided to accept the assertion in return for being able to immediately flag this when introducing the error, where it is likely to be seen. If only we had custom type errors in TS. I'll undo.

![image](https://user-images.githubusercontent.com/44588767/197507253-9824ea23-4589-4c1f-8f18-bbda9e777d67.png)

Reply:
we should revisit stuff like this at some point. I like the idea of explicit and clear compile-time errors.